src/Chainweb/Pact/Backend/DbCache.hs

```diff
@@ -168,7 +173,7 @@ checkDbCache key rowdata txid = runStateT $ do
     readCache txid addy >>= \case
 
         -- Cache hit
-        Just x -> do
+        Just !x -> do
```

Comment:
I would prefer if instead of forcing on read, we force on write. We could add a bang to the `v` parameter of `writeCache`.